### PR TITLE
gps: improve error handling

### PR DESCRIPTION
**Pull request by deadprogram**
This PR makes some improvements to how the `gps` driver handles errors. There were some edge cases before which could cause a panic. Also, some errors now return more information about what caused the error.

### CODE REVIEW

File: gps/gps.go [hunk 1]
```diff
@@ -127,16 +154,18 @@ func (gps *Device) WriteBytes(bytes []byte) {
 
 // validSentence checks if a sentence has been received uncorrupted
 func validSentence(sentence string) error {
-	if len(sentence) < 4 || sentence[0] != '$' || sentence[len(sentence)-3] != '*' {
+	if len(sentence) < 7 || sentence[0] != '$' || sentence[len(sentence)-3] != '*' {
```
Comment (ysoldak) [inline gps/gps.go]: What are magic indexes here? Add a comment about the structure of the sentence?

File: gps/gpsparser.go [hunk 2]
```diff
@@ -100,7 +112,10 @@ func findTime(val string) time.Time {
 	h, _ := strconv.ParseInt(val[0:2], 10, 8)
 	m, _ := strconv.ParseInt(val[2:4], 10, 8)
 	s, _ := strconv.ParseInt(val[4:6], 10, 8)
-	ms, _ := strconv.ParseInt(val[7:10], 10, 16)
+	ms := int64(0)
+	if len(val) == 10 {
```
Comment (ysoldak) [inline gps/gpsparser.go]:
```suggestion
	if len(val) >= 10 {
```
It's OK for `val` to be longer or we are working with just the block between two first commas here?

File: gps/gps.go (hunk 1 above)
Comment (deadprogram) [reply gps/gps.go]: Added some const and comments.

File: gps/gpsparser.go (hunk 2 above)
Comment (deadprogram) [reply gps/gpsparser.go]: Improved that slice expression, good catch!

**Comment by deadprogram** [comment]
Added some unit tests, as well.

**Comment by ofauchon** [comment]
I ran first 'dev' branch of the driver on Beitian BN 220 GPS module on STM32WL board.

And I quickly got a panic: 

```
$GNTXT,01,01,02,HW UBX-M8030 00080000*60
$GNTXT,01,01,02,FWVER=SPG 3.01*46
$GNTXT,01,01,02,PROTVER=18.00*11
$GNTXT,01,01,02,GPS;GLO;GAL;BDS*77
$GNTXT,01,01,02,SBAS;IMES;QZSS*49
$GNTXT,01,01,02,GNSS OTP=GPS;GLO*37
$GNRMC,212752.10,V,,,,,,,160123,,,N*64
panic: runtime error: slice out of range
```

Then the 'improve-gps-parsing' branch, which worked properly : 

```
$GNRMC,212838.86,V,,,,,,,160123,,,N*68
$GNVTG,,,,,,,,,N*2E
$GNGGA,212838.86,,,,,0,00,99.99,,,,,,*74
$GPGSV,4,4,14,32,06,130,,36,33,144,*7D
$GLGSV,2,1,06,69,21,033,,70,39,104,,71,14,152,,76,41,279,*69
$GLGSV,2,2,06,84,07,077,,85,55,033,*65
$GNGLL,,,,,212839.00,V,N*57
$GNTXT,01,01,02,ANTSTATUS=INIT*3B
$GNRMC,212840.00,V,,,,,,,160123,,,N*69
$GNVTG,,,,,,,,,N*2E
$GNGGA,212840.00,,,,,0,00,99.99,,,,,,*75
$GNGSA,A,1,,,,,,,,,,,,,99.99,99.99,99.99*2E
GOT FIX ! xxxxx
GOT FIX ! xxxxx
```

Seems the previous version of the driver (dev branch), can't process '$GNVTG,,,,,,,,,N*2E' frame

Unknown frames fires correct errors:

```
unsupported NMEA sentence type GSV $GPGSV,4,1,13,01,10,255,,07,05,285,,08,61,311,,10,61,082,25*76
unsupported NMEA sentence type GSV $GPGSV,4,2,13,16,40,189,22,18,06,067,,21,35,266,,23,33,049,*7B
unsupported NMEA sentence type GSV $GPGSV,4,3,13,26,10,173,,27,79,087,24,30,02,318,,32,11,126,*71
unsupported NMEA sentence type GSV $GPGSV,4,4,13,36,33,144,*4F
unsupported NMEA sentence type GSV $GLGSV,2,1,08,69,17,027,,70,43,095,,71,21,150,,76,38,269,*68
```

Thanks @deadprogram for this improvement.

**Comment by deadprogram** [comment]
Thanks for testing/review @ofauchon and @ysoldak now merging.

**Pull request merged**